account_banking_pain_base/models/res_company.py

```diff
@@ -62,19 +45,82 @@ def _initiating_party_issuer_default(self, cr, uid, context=None):
+
+    initiating_party_issuer = fields.Char(
+        'Initiating Party Issuer', size=35,
+        help="This will be used as the 'Initiating Party Issuer' in the "
+        "PAIN files generated by Odoo.", default=_initiating_party_issuer_def)
+    initiating_party_identifier = fields.Char(
+        string='SEPA Initiating Party Identifier', size=35,
+        help="Enter the SEPA Initiating Party Identifier that has been "
+             "attributed to your company to make SEPA operations. This "
+             "identifier is composed of :\n- your country ISO code (2 "
+             "letters)\n- a 2-digits checkum\n- a 3-letters business "
+             "code\n- a country-specific identifier",
+        oldname="sepa_creditor_identifier")
+
+    @api.model
+    def _get_initiating_party_identifier(self, company_id):
+        """The code here may be different from one country to another.
+        If you need to add support for an additionnal country, you can
+        contribute your code here or inherit this function in the
+        localization modules for your country"""
+        assert isinstance(company_id, int), 'Only one company ID'
+        company = self.browse(company_id)
+        company_vat = company.vat
+        party_identifier = company.initiating_party_identifier
+        if not party_identifier and company_vat:
+            country_code = company_vat[0:2].upper()
+            if country_code == 'BE':
+                party_identifier = company_vat[2:].replace(' ', '')
+        return party_identifier
+
+    def is_initiating_party_identifier_valid(
+            self, initiating_party_identifier):
+        """Check if SEPA Creditor Identifier is valid
+        @param initiating_party_identifier: SEPA Initiating Party Identifier
+            as str or unicode
+        @return: True if valid, False otherwise
+        """
+        if not isinstance(initiating_party_identifier, (str, unicode)):
+            return False
+        try:
+            ipi = str(initiating_party_identifier).lower()
+        except:
+            logger.warning(
+                "SEPA Creditor ID should contain only ASCII caracters.")
+            return False
+        if len(ipi) < 9:
+            return False
+        before_replacement = ipi[7:] + ipi[0:2] + '00'
+        logger.debug(
+            "SEPA ID check before_replacement = %s" % before_replacement)
+        after_replacement = ''
+        for char in before_replacement:
+            if char.isalpha():
+                after_replacement += str(ord(char) - 87)
+            else:
+                after_replacement += char
+        logger.debug(
+            "SEPA ID check after_replacement = %s" % after_replacement)
+        return int(ipi[2:4]) == (98 - (int(after_replacement) % 97))
 
-    _defaults = {
-        'initiating_party_issuer': _initiating_party_issuer_def,
-    }
+    @api.one
+    @api.constrains('initiating_party_identifier')
+    def _check_initiating_party_identifier(self):
```

Comment:
should be _check_sepa_creditor_identifier and go into account_banking_sepa_direct_debit